tests/test_velodyne_points_format.py

```diff
@@ -0,0 +1,245 @@
+                                         "path": self.image1},
+                                        ],
+                        attributes={'frame': 40}
+                        )
+        ], categories={
+            AnnotationType.label: LabelCategories.from_iterable(['car'])
+        })
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(expected_dataset,
+                VelodynePointsConverter.convert, test_dir, **self.dimension)
+
+    def test_reindex(self):
+        source_dataset = Dataset.from_iterable([
+            DatasetItem(id='frame_000000',
+                        annotations=[Cuboid(id=0,
+                                            attributes={'occluded': 0},
+                                            group=0,
+                                            points=[13.54034048060633, -9.410120134481405, 0.24972983624747647, 0.0,
+                                                    0.0, 0.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                            label=0, z_order=0)],
+                        subset='tracklets',
+                        path=[],
+                        image=None,
+                        pcd=self.pcd1,
+                        related_images=[{"name": "0000000000.png", "save_path": "IMAGE_00",
+                                         "path": self.image1},
+                                        ],
+                        attributes={'frame': 20}
+                        )
+        ], categories={
+  AnnotationType.label: LabelCategories.from_iterable(['car'])
+})
+
+        expected_dataset = Dataset.from_iterable([
+            DatasetItem(id='frame_000000',
+                        annotations=[Cuboid(id=0,
+                                            attributes={'occluded': 0},
+                                            group=0,
+                                            points=[13.54034048060633, -9.410120134481405, 0.24972983624747647, 0.0,
+                                                    0.0, 0.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                            label=0, z_order=0)],
+                        subset='tracklets',
+                        path=[],
+                        image=None,
+                        pcd=self.pcd1,
+                        related_images=[{"name": "0000000002.png", "save_path": "IMAGE_00",
+                                         "path": self.image1},
+                                        ],
+                        attributes={'frame': 0}
+                        )
+                ], categories={
+          AnnotationType.label: LabelCategories.from_iterable(['car'])
+        })
+
+        with TestDir() as test_dir:
+            self._test_save_and_load(source_dataset,
+                partial(VelodynePointsConverter.convert, reindex=True), test_dir,
+                target_dataset=expected_dataset, **self.dimension)
+
```

Comment:
Would other typical format tests be valid for these formats? Relative image paths, arbitrary attributes, multiple subsets, no subsets, no images (only annotations)?

Reply:
Only tests for without images is missing, we shall add that

Reply:
Updated tests